plan/plan.go

```diff
@@ -265,7 +265,7 @@ func (p *Plan) Calculate() *Plan {
 		Current:        p.Current,
 		Desired:        p.Desired,
 		Changes:        changes,
-		ManagedRecords: []string{endpoint.RecordTypeA, endpoint.RecordTypeAAAA, endpoint.RecordTypeCNAME},
+		ManagedRecords: p.ManagedRecords,
```

Comment:
@Raffo All the tests pass, but I still have doubts about this one.

Reply:
Right. Now I understand your concerns. It looks like a breaking change to me, if you build and run this pull request.

At the moment 
```
| `--managed-record-types=A...` | Record types to manage; specify multiple times to include many; (default: A, AAAA, CNAME) (supported records: A, AAAA, CNAME, NS, SRV, TXT) |
```

With this change, everyone has to add `A, AAAA, CNAME` with the flag explicitly.

Reply:
I think we should have a constant or something, so that if we change value, documentation for flags changed as well

Now
```
app.Flag("managed-record-types", "Record types to manage; specify multiple times to include many; (default: A, AAAA, CNAME) (supported records: A, AAAA, CNAME, NS, SRV, TXT)").Default("A", "AAAA", "CNAME").StringsVar(&cfg.ManagedDNSRecordTypes)
```

To
```
app.Flag("managed-record-types", "Record types to manage; specify multiple times to include many; (default: A, AAAA, CNAME) (supported records: A, AAAA, CNAME, NS, SRV, TXT)").Default(DefaultManagedRecordTypes).StringsVar(&cfg.ManagedDNSRecordTypes)
```

And the flag value should be passed/propaged to the plan

Reply:
That makes sense. I'll see what I can do about it.

Reply:
I re-ordered alphabetically defaultConfig and so, I was able to see that there were already one value for that. 
So I used it instead of the multiple hard-coded values.

I followed the code:
* in `controller/execute.go:350`, it sets the ManagedDNSRecordsTypes from the config into _Controller_.
* in `controller/controller.go:250`, it sets the _Plan_ with ManagedDNSRecordsTypes from this _Controller

So my guess is that it should work. I'll test it when I can.

Reply:
@mloiseleur this is extremely intentional. The default for ExternalDNS was to always only consider A/AAAA and CNAMEs and everything else is an add on or something to be considered. Remember that ExternalDNS was built with the idea of the “PaaS use case”, as a tool to give you easily a DNS record for when you deploy an application, the other use cases are additions that are not technically part of the core. I would be very careful of introducing a breaking change here.

Reply:
Thanks. I'll revert this specific change then, and add a comment explaining why it's hard coded here.